hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashDeletingService.java

```diff
@@ -0,0 +1,114 @@
+import org.apache.hadoop.hdds.utils.BackgroundService;
+import org.apache.hadoop.hdds.utils.BackgroundTask;
+import org.apache.hadoop.hdds.utils.BackgroundTaskQueue;
+import org.apache.hadoop.hdds.utils.BackgroundTaskResult;
+import org.apache.hadoop.ozone.OzoneConsts;
+import org.apache.hadoop.security.SecurityUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.security.PrivilegedExceptionAction;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.hadoop.ozone.om.OMConfigKeys.OZONE_OM_ADDRESS_KEY;
+
+/**
+ * Background Service to empty keys that are moved to Trash.
+ */
+public class TrashDeletingService extends BackgroundService {
+
+    private static final Logger LOG =
+            LoggerFactory.getLogger(TrashDeletingService.class);
+
+    // Use single thread for  now
+    private final static int KEY_DELETING_CORE_POOL_SIZE = 1;
+
+    private OzoneManager ozoneManager;
+
+    public void setFsConf(Configuration fsConf) {
+        this.fsConf = fsConf;
+    }
+
+    private Configuration fsConf;
+
+    public TrashDeletingService(long interval, long serviceTimeout,OzoneManager ozoneManager) {
+        super("TrashDeletingService", interval, TimeUnit.MILLISECONDS, KEY_DELETING_CORE_POOL_SIZE, serviceTimeout);
+        this.ozoneManager = ozoneManager;
+        fsConf = new Configuration();
+    }
+
+
+    @Override
+    public BackgroundTaskQueue getTasks() {
+        BackgroundTaskQueue queue = new BackgroundTaskQueue();
+        String rootPath = String.format("%s://%s/",
+                OzoneConsts.OZONE_OFS_URI_SCHEME, ozoneManager.getConfiguration().get(OZONE_OM_ADDRESS_KEY));
+        // Configuration object where the rootpath is set to an OFS Uri.
+        fsConf.set(CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY, rootPath);
+        FileSystem fs = null;
+        try {
+            fs = SecurityUtil.doAsLoginUser(
+                    new PrivilegedExceptionAction<FileSystem>() {
+                        @Override
+                        public FileSystem run() throws IOException {
+                            return FileSystem.get(fsConf);
+                        }
+                    });
+        } catch (IOException e) {
+            LOG.error("Cannot instantiate filesystem instance");
+        }
+        queue.add(new TrashDeletingTask(fs,fsConf));
+        return queue;
+    }
+
+    /**
+     * This task creates an emptier thread that deletes all keys obtained from the trashRoots (fs.getTrashRoots)
+     */
+    private class TrashDeletingTask implements BackgroundTask {
+
+        FileSystem fs;
+        Configuration conf;
+
+        public TrashDeletingTask(FileSystem fs, Configuration conf) {
+            this.fs = fs;
+            this.conf = conf;
+        }
+
+        @Override
+        public BackgroundTaskResult call() throws Exception {
+            Thread emptier = new Thread(new Trash(fs,conf).getEmptier(),"Trash Emptier");
```

Comment:
This will start 1 thread for the entire filesystem for every run? why do we need this a background service ?